lms/djangoapps/commerce/views.py

```diff
@@ -103,28 +103,26 @@ def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
 
         # Make the API call
         try:
-            order_number, order_status, _body = api.create_order(user, honor_mode.sku)
-            if order_status == OrderStatus.COMPLETE:
-                msg = Messages.ORDER_COMPLETED.format(order_number=order_number)
+            response_data = api.create_basket(
+                user,
+                honor_mode.sku,
+                payment_processor="cybersource",
+            )
+            if response_data['payment_data'] is not None:
+                # it is time to start the payment flow.
+                return JsonResponse(response_data['payment_data'])
+            elif response_data['order']:
+                # the order was completed immediately because there was no charge.
+                msg = Messages.ORDER_COMPLETED.format(order_number=response_data['order']['number'])
                 log.debug(msg)
                 return DetailResponse(msg)
             else:
-                # TODO Before this functionality is fully rolled-out, this branch should be updated to NOT enroll the
-                # user. Enrollments must be initiated by the E-Commerce API only.
-                self._enroll(course_key, user)
```

Comment:
Since you raise this, after some consideration, I'm going to put it back.  I was originally thinking this failsafe wasn't going to be a good idea when the mode being handled is a paid mode.  I've since become clear on the fact that this endpoint doesn't have to handle paid modes yet.